src/testrel.jl

```diff
@@ -432,33 +438,39 @@ function _test_rel_steps(;
         end
         duration = sprint(show, stats.time; context=:compact => true)
         ts = stats.value
-        ts.logs = logger.logs
 
-        check_flaky(name, logger.logs)
+        # If we're capturing logs then we need to make an exception to report error states
+        if capture_logs
+            ts.logs = logger.logs
+            check_flaky(name, logger.logs)
+
+            log_header = get_log_header(ts, duration, schema, test_engine)
+            if anyerror(ts) || anyfail(ts)
+                ts.error_message = log_header
+                io, ctx = get_logging_io()
+                write(ctx, log_header)
+                write(ctx, "\n\nCAPTURED LOGS:\n")
+                playback_log.(ctx, logger.logs)
+                msg = String(take!(io))
+                @error msg database = schema engine_name = test_engine
+            else
+                @info log_header
+            end
+        end
 
-        log_header = get_log_header(ts, duration, schema, test_engine)
-        if anyerror(ts) || anyfail(ts)
-            ts.error_message = log_header
+        return ts
+    catch err
+        if capture_logs
             io, ctx = get_logging_io()
-            write(ctx, log_header)
-            write(ctx, "\n\nCAPTURED LOGS:\n")
+            write(ctx, "[ERROR] Something went wrong running test $name \n\n CAPTURED LOGS:\n")
+
+            # dump all of the captured logs
             playback_log.(ctx, logger.logs)
+            Base.show(ctx, err)
             msg = String(take!(io))
-            @error msg database = schema engine_name = test_engine
         else
-            @info log_header
+            msg = ""
         end
```

Comment:
instead of these changes, could we do a smaller (but less lively) intervention where we add an `elseif` branch here that checks if we're not nested and then just plays back the logs immediately?

Reply:
The immediacy is lost, but it would be nicer in most other ways. I was also thinking of tying it to `debug` instead of nesting.

Reply:
or either nesting or debug? I'm fine with losing the immediacy for the simplicity of code change it allows.

If we need immediacy, I'd probably think about making a custom logger that wraps the test logger but keeps track of whether to log things to the current logger when we get an incoming message to handle.